### PR TITLE
Add/update PSRAM config to sdkconfig

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add options to allow proper detection on boot.
- Remove mem test options.
- Add config to enable PSRAM support on default sdkconfig.

## Motivation and Context
- Option to init SPIRAM on boot was missing to allow detection in nanoff (see nanoframework/nanoFirmwareFlasher#270).
- No need to perform memory test on boot.
- Add missing configs to enable PSRAM support on default sdkconfig.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

## Summary by CodeRabbit

- **New Features**
  - Added SPIRAM configuration options to enhance memory management and initialization.
  - Modified partition table customization for better flexibility and control.

- **Improvements**
  - Enhanced system behavior during boot and memory allocation by enabling SPIRAM boot initialization.
  - Improved memory allocation for WiFi LWIP by attempting to use SPIRAM.

- **Bug Fixes**
  - Configured the system to ignore missing SPIRAM to prevent boot issues.

- **Removals**
  - Removed redundant SPIRAM memory test configurations to streamline the boot process.

### CODE REVIEW

**Review by Ellerbach — APPROVED** [review]
Seems like a great finding!

File: targets/ESP32/_IDF/sdkconfig.default [hunk 1]
```diff
@@ -8,6 +8,10 @@ CONFIG_IDF_FIRMWARE_CHIP_ID=0x0000
 CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
 CONFIG_ESPTOOLPY_FLASHSIZE_4MB=y
 
+CONFIG_SPIRAM=y
```
Comment (AdrianSoundy) [inline targets/ESP32/_IDF/sdkconfig.default:11]: Surprised this wasn't set as its used in a number of configs that have psram.  WROVER_KIT, ESP32_PSRAM_REV0 & ESP32_PSRAM_XTAL26_REV0
Also others that don't have psram

Didn't seem to be set in IDF 4 version either

**Review by AdrianSoundy — APPROVED** [review]
LGTM
and not a IDF 5 problem 👍

File: targets/ESP32/_IDF/sdkconfig.default (hunk 1 above)
Comment (josesimoes) [reply targets/ESP32/_IDF/sdkconfig.default:11]: Yeah.. haven't trace it back to IDF 4 but it is what it is...

**Pull request merged**